### ISSUE TITLE
Frame rate limits ?

### ISSUE THREAD

**Issue by JimmyDaSilva** [Sonnentierchen/Qt3D-OfflineRenderer#2]
Hi @Sonnentierchen,
thank you for your implementation. you did save me a lot of time !

I am experiencing differences in frame rate (smoothness of changes in the image) when I compare the display inside the offscreen renderer and the basic view of Qt3DWindow.
I would like to know if there is any limitations in your approach that makes it impossible to be as fast as Qt3DWindow.

I am new to 3D rendering and I am having trouble understanding why we can't, somehow, just get directly the image shown on the screen at the speed the renderer renders the 2D view of the scene.

Thanks again for sharing this =)

**Comment by florianblume** [comment]
Hi @JimmyDaSilva,

I'm glad I could help you! I implemented this as part of my university project and the lack of decent documentation of the classes and the Qt3D framework in general drove me insane that's why I shared this in the first place and am so happy to having spared someone the troubles and annoyance I had to endure :D

Unfortunately, I'm not sure why this is the case. I tried to peek into the implementation details, but you only get to see the front-end parts and not the back-end of the framework that actually translates the nodes into OpenGL commands. But my wildest guess is that is simply and issue of the framework itself. If you had a look into the stop_engine branch and tried it out you can see that the application crashes when you set the root node of the QAspectEngine to an empty pointer. According to the documentation this should stop the graphics loop. Besides, the CPU load is constantly wierdly high, at least on my Core i5 dual core (about 15% when the onscreen rendering takes about 3%). You can't notice this on a more powerful machine but this should definitely not be the case.

So to summarize: I have no idea :D But it looks like the whole 3D framework has not yet been fully developed and is undergoing a lot of optimization, so maybe we just have to wait...

**Comment by JimmyDaSilva** [comment]
@Sonnentierchen Thank you for your personal feedback.
For my part your program runs with 5% load on my 4-cores i7, and basic-shapes runs with 2%. So I observe the same problem.

I hope qt3d advancements will help soon !
